src/js/core/widget/core/Spin.js

```diff
@@ -0,0 +1,630 @@
+				window.setTimeout(function () {
+					element.classList.remove(classes.ENABLING);
+					self.refresh();
+				}, ENABLING_DURATION);
+				element.classList.remove(classes.ENABLED);
+				utilsEvents.off(document, "drag dragend", self);
+				// disable animation
+				self._animation.stop();
+				// enable tau rotaryScroller the widget has own support for rotary event
+				ns.util.rotaryScrolling && ns.util.rotaryScrolling.unlock();
+			}
+			// reset previous value;
+			this._prevValue = null;
+			return true;
+		};
+
+		prototype._setDirection = function (element, direction) {
+			this.options.direction = (["up", "down"].indexOf(direction) > -1) ?
+				direction : "up";
+		};
+
+		prototype._drag = function (e) {
+			var self = this,
+				dragValue,
+				value;
+
+			// if element is detached from DOM then event listener should be removed
+			if (document.getElementById(self.element.id) === null) {
+				utilsEvents.off(document, "drag dragend", self);
+			} else {
+				if (self.options.enabled) {
+					value = self.value();
+					dragValue = e.detail.deltaY - lastDragValueChange;
+
+					if (Math.abs(dragValue) > DRAG_STEP_TO_VALUE) {
+						self._setValue(value - Math.round(dragValue / DRAG_STEP_TO_VALUE), true);
+						lastDragValueChange = e.detail.deltaY;
+						window.navigator.vibrate(VIBRATION_DURATION);
+					}
+				}
+			}
+
+		};
+
+		prototype._dragEnd = function () {
+			lastDragValueChange = 0;
+		};
+
+		prototype._click = function (e) {
+			var target = e.target,
+				self = this,
+				items = self._ui.items,
+				value = self.value(),
+				targetIndex = items.indexOf(target),
+				currentIndex = self._valueToIndex(value);
+
+			if (targetIndex > -1 && targetIndex !== currentIndex) {
+				if (currentIndex === items.length - 1 && targetIndex == 0) {
+					// loop - current index is 12/12 and event target has index 0/12
+					self._setValue(value + 1, true);
+				} else if (currentIndex === 0 && targetIndex == items.length - 1) {
+					// loop - current index is 0/12 and event target has index 12/12
+					self._setValue(value - 1, true);
+				} else if (targetIndex < currentIndex) {
+					self._setValue(value - 1, true);
+				} else if (targetIndex > currentIndex) {
+					self._setValue(value + 1, true);
+				}
+			}
+		}
+
+		prototype.handleEvent = function (event) {
+			switch (event.type) {
+				case "drag":
+					this._drag(event);
+					break;
+				case "dragend":
+					this._dragEnd(event);
+					break;
+				case "click":
+					this._click(event);
+					break;
+			}
+		};
+
+		prototype._bindEvents = function () {
+			var self = this;
+
+			// enabled drag gesture for document
+			utilsEvents.enableGesture(document, new gesture.Drag({
+				blockHorizontal: true
+			}));
+
+			utilsEvents.on(self.element, "click", self);
+		};
+
+		prototype._unbindEvents = function () {
+			var self = this;
+
+			utilsEvents.disableGesture(document);
+
+			utilsEvents.off(document, "drag dragend", self);
```

Comment:
just question.
can drag and dragend be handled in one off?

Reply:
Yes. Spin turn on the "drag dragend" in the 491 line.
So I try to turn off the "drag dragend" events.